.github/workflows/canister-tests.yml

```diff
@@ -0,0 +1,292 @@
+
+      - name: 'Download wasm'
+        uses: actions/download-artifact@v2
+        with:
+          name: internet_identity_dummy_captcha.wasm
+          path: .
+
+      - name: Run Tests
+        shell: bash
+        working-directory: backend-tests
+        run: cabal run
+
+  ######################
+  # The selenium tests #
+  ######################
+
+  selenium:
+    runs-on: ubuntu-latest
+    needs: docker-build-dummy-captcha
+    strategy:
+      matrix:
+        start-flag: [ '', '--emulator' ]
+        device: [ 'desktop', 'mobile' ]
+      # Make sure that one failing test does not cancel all other matrix jobs
+      fail-fast: false
+    env:
+      DFX_VERSION: 0.8.3
+      REPLICA_URL: https://ic0.app
+      II_ORIGIN: https://identity.ic0.app
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: 14.x
+
+      # This step hangs on Github actions on Darwin for some reason, that
+      # is why we run this only on Linux for now
+      - name: Install DFX
+        run: |
+          echo Install DFX Version: "$DFX_VERSION"
+          yes | sh -ci "$(curl -fsSL https://sdk.dfinity.org/install.sh)"
+
+      # Helps with debugging
+      - name: Show versions
+        run: |
+          echo dfx --version
+          dfx --version
+          echo node --version
+          node --version
+          echo docker --version
+          docker --version
+
+      # needed so that we can run the test on identity.ic0.app
+      - name: Add hosts to /etc/hosts
+        run: |
+          sudo echo "127.0.0.1 identity.ic0.app" | sudo tee -a /etc/hosts
+          sudo echo "127.0.0.1 ic0.app" | sudo tee -a /etc/hosts
+
+      - run: docker build reverse_proxy -t ii_proxy
+      - run: docker run --add-host host.docker.internal:host-gateway -d -p 443:443 ii_proxy
+      - run: dfx start ${{ matrix.start-flag }} --background
+
+      - name: 'Download wasm'
+        uses: actions/download-artifact@v2
+        with:
+          name: internet_identity_dummy_captcha.wasm
+          path: .
+
+      - name: Deploy Internet Identity
+        run: |
+          dfx canister --no-wallet create --all
+          dfx canister --no-wallet install internet_identity --argument '(null)'
+
+      - name: Deploy whoami canister
+        run: cd demos/whoami && dfx deploy --no-wallet
+
+      # We don't use the canister in the tests, but we want to make sure it
+      # can be built and deployed
+      - name: Deploy testnet_app canister
+        run: |
+          cd demos/testnet_app
+          npm ci
+          dfx deploy --no-wallet
+
+      - name: Run demo app
+        working-directory: demos/sample-javascript
+        run: |
+          npm ci
+          npm run build
+          npm run develop --no-live-reload --no-devtool &
+          sleep 10 # crude
+
+      - name: install selenium webdrivers
```

Comment:
Is this running from the same working directory as the previous step?

Reply:
nope it's not! each step runs in the (repo) root directory by default